detection_rules/rule.py

```diff
@@ -714,6 +719,20 @@ def interval_ratio(self) -> Optional[float]:
             return interval / self.max_span
 
 
+@dataclass(frozen=True)
+class ESQLRuleData(QueryRuleData):
+    """ESQL rules are a special case of query rules."""
+    type: Literal["esql"]
+    language: Literal["esql"]
+    query: str
+
+    @validates_schema
+    def validates_esql_data(self, data, **kwargs):
+        """Custom validation for query rule type and subclasses."""
+        if data.get('index'):
```

Comment:
@Mikaayenson @terrancedejesus - index invalidation

Reply:
I think we may want to inherit `BaseRuleData` and not raise an error on finding index. We often treat `QueryRuleData` or the Query rule type as base, which diverges from the schema design upstream as we then need to make exceptions, this is an example, when we inherit into other rule types. Every rule type, except for ML, inherits QueryRuleData. Also, since the dataclasses are immutable we may want to stray away from any dataclass post loading validation or changes.